pkg/controller/common/util.go

```diff
@@ -917,6 +936,15 @@ func SetRestrictedSecurityContext(podSpec *v1.PodSpec) {
 	}
 }
 
+// SetNodeNameIfPopulator sets NodeName in a pod spec when the PVC is being handled by a CDI volume populator
+func SetNodeNameIfPopulator(pvc *corev1.PersistentVolumeClaim, podSpec *v1.PodSpec) {
+	_, isPopulator := pvc.Annotations[AnnPopulatorKind]
+	nodeName, _ := pvc.Annotations[AnnSelectedNode]
+	if isPopulator && nodeName != "" {
```

Comment:
So I previously tried other two things:
- Having an additional annotation in the PVC Prime  (`AnnSelectedNodeName`) and just copy that one into the pod NodeName. This failed because the populator controller needs to also set `AnnSelectedNode` into PVC' for it to work.
- Always copying `AnnSelectedNode` into the pod spec's NodeName. This caused lots of errors in the tests when a PVC created with the regular import flow had `AnnSelectedNode` (the DV stayed in ImportScheduled until timeout).

So I'm just copying AnnSelectedNode when the import-controller reconciles a PVC', which seems to work and doesn't mess with the standard import flow while avoiding creating another redundant annotation.

Reply:
I think it is unfortunate for this layer to know anything about populators.

Reply:
I agree but having another annotation exclusive to populators with the same info as AnnSelectedNode seemed bad too. I can change it though.